File: frontend/src/api/operations.ts
import { api } from "./client";

export interface OnCallEngineer {
  name: string;
  email: string;
  shift_start: string;
  shift_end: string;
}

export interface SloWindow {
  service: string;
  availability: number;
  latency_ms: number;
  window_start: string;
  window_end: string;
}

export interface MaintenanceEvent {
  id: string;
  description: string;
  scheduled_for: string;
  impact: string;
}

export interface OperationsOverview {
  on_call: OnCallEngineer[];
  slos: SloWindow[];
  maintenance: MaintenanceEvent[];
}

export const fetchOperationsOverview = async (): Promise<OperationsOverview> => {
  const { data } = await api.get<OperationsOverview>("/operations/overview");
  return data;
};

File: backend/app/schemas/operations.py
from dataclasses import dataclass
from datetime import datetime
from typing import List


@dataclass
class OnCallEngineer:
    name: str
    email: str
    shift_start: datetime
    shift_end: datetime


@dataclass
class SloWindow:
    service: str
    availability: float
    latency_ms: float
    window_start: datetime
    window_end: datetime


@dataclass
class MaintenanceEvent:
    id: str
    description: str
    scheduled_for: datetime
    impact: str


@dataclass
class OperationsOverview:
    on_call: List[OnCallEngineer]
    slos: List[SloWindow]
    maintenance: List[MaintenanceEvent]
